src/identity/IdentityProviderHttpHandler.ts

```diff
@@ -0,0 +1,90 @@
+// it to be possible, but we must get the configuration to satisfy the needs of the CSS
+// architecture. See the "canHandle" method for an explantaion
+// eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error, @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import instance from 'oidc-provider/lib/helpers/weak_cache';
+import type { HttpHandlerInput } from '../server/HttpHandler';
+import { HttpHandler } from '../server/HttpHandler';
+import { NotImplementedHttpError } from '../util/errors/NotImplementedHttpError';
+import type { IdentityProviderFactory } from './IdentityProviderFactory';
+import type { IdPInteractionPolicyHttpHandler } from './interaction/IdPInteractionPolicyHttpHandler';
+
+export class IdentityProviderHttpHandler extends HttpHandler {
+  private readonly provider: Provider;
+  private readonly interactionPolicyHttpHandler: IdPInteractionPolicyHttpHandler;
+
+  public constructor(
+    providerFactory: IdentityProviderFactory,
+    interactionPolicyHttpHandler: IdPInteractionPolicyHttpHandler,
+  ) {
+    super();
+    this.interactionPolicyHttpHandler = interactionPolicyHttpHandler;
+    this.provider = providerFactory.createProvider(this.interactionPolicyHttpHandler);
+  }
+
+  /**
+   * Handles a request. Returns a promise that will either resolve if a response is
+   * given (including if the response is an error page) and throw an error if the
+   * idp cannot handle the request.
+   * NOTE: This method has a lot of hacks in it to get it to work with node-oidc-provider.
+   */
+  public async canHandle(input: HttpHandlerInput): Promise<void> {
+    // Get the routes from the configuration. `instance` is needed because the configuration
+    // is not actually stored in the provider object, but rather in a WeakMap accessed by
+    // the provider instance.
+    // https://github.com/panva/node-oidc-provider/blob/master/lib/provider.js#L88-L91
+    const routesMap: Record<string, string> = instance(this.provider).configuration().routes;
+    const validRoutes: string[] = Object.values(routesMap);
+    validRoutes.push('/.well-known/openid-configuration');
+    // URL.parse is deprecated, but as of the time this comment was written, there
+    // doesn't seem to be a perfect replacement for it, so we will keep it until there
+    // is a perfect replacement
+    // https://github.com/nodejs/node/issues/12682#issuecomment-736510378
+    const url = input.request.url ? parse(input.request.url).pathname as string : '';
+
+    let interactionHttpHandlerCanHandle = true;
+    try {
+      await this.interactionPolicyHttpHandler.canHandle({ ...input, provider: this.provider });
+    } catch {
+      interactionHttpHandlerCanHandle = false;
+    }
+
+    // Account for special case where the interaction policy calls back to "/auth/:uid"
+    const isSpecialCallbackRoute =
+      new RegExp(
+        `^${routesMap.authorization}/[_A-Za-z0-9\\-]+/?$`,
```

Comment:
This needs escaping. See `escape-string-regexp` lib we are already using.

Reply:
Done